include/utils.h

```diff
@@ -3,9 +3,28 @@
 
 #include <err.h>
 #include <stdlib.h>
-
+#include <stdint.h>
+/**
+ * Mark a function as unimplemented and error out when called
+ */
 #define UNIMPLEMENTED()                                                        \
-	errx(EXIT_FAILURE, "function %s in %s:%d not implemented",             \
-	     __FUNCTION__, __FILE__, __LINE__)
+	errx(EXIT_FAILURE, "function %s in %s:%d not implemented", __func__,   \
+	     __FILE__, __LINE__)
+
+/**
+ * Mark argument as `unused`, for example when developing unimplemented functions
+ */
+#define UNUSED(__ARG) __attribute__((__unused__)) __ARG
+
+/**
+ * Create a unique phantom member to a struct to avoid empty structs.
```

Comment:
why do we need to avoid empty struct ?

Reply:
ISO C does not allow empty structs

Reply:
ISO C always doesn't have `__FUNCTION__` do we really want ISO C ?